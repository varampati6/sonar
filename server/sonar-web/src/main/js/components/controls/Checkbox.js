/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import classNames from 'classnames';

export default class Checkbox extends React.PureComponent {
  static propTypes = {
    id: React.PropTypes.string,
    onCheck: React.PropTypes.func.isRequired,
    checked: React.PropTypes.bool.isRequired,
    thirdState: React.PropTypes.bool,
    className: React.PropTypes.any
  };

  static defaultProps = {
    thirdState: false
  };

  componentWillMount() {
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    e.preventDefault();
    e.target.blur();
    this.props.onCheck(!this.props.checked, this.props.id);
  }

  render() {
    const className = classNames('icon-checkbox', this.props.className, {
      'icon-checkbox-checked': this.props.checked,
      'icon-checkbox-single': this.props.thirdState
    });

    return <a id={this.props.id} className={className} href="#" onClick={this.handleClick} />;
  }
}
