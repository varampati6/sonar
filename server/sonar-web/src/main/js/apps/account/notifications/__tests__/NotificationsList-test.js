/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import NotificationsList from '../NotificationsList';
import Checkbox from '../../../../components/controls/Checkbox';

const channels = ['channel1', 'channel2'];
const types = ['type1', 'type2'];
const notifications = [
  { channel: 'channel1', type: 'type1' },
  { channel: 'channel1', type: 'type2' },
  { channel: 'channel2', type: 'type2' }
];
const checkboxId = (t, c) => `checkbox-io-${t}-${c}`;

it('should match snapshot', () => {
  expect(
    shallow(
      <NotificationsList
        onAdd={jest.fn()}
        onRemove={jest.fn()}
        channels={channels}
        checkboxId={checkboxId}
        types={types}
        notifications={notifications}
      />
    )
  ).toMatchSnapshot();
});

it('should call `onAdd` and `onRemove`', () => {
  const onAdd = jest.fn();
  const onRemove = jest.fn();
  const wrapper = shallow(
    <NotificationsList
      onAdd={onAdd}
      onRemove={onRemove}
      channels={channels}
      checkboxId={checkboxId}
      types={types}
      notifications={notifications}
    />
  );
  const checkbox = wrapper.find(Checkbox).first();

  checkbox.prop('onCheck')(true);
  expect(onAdd).toHaveBeenCalledWith({ channel: 'channel1', type: 'type1' });

  jest.resetAllMocks();

  checkbox.prop('onCheck')(false);
  expect(onRemove).toHaveBeenCalledWith({ channel: 'channel1', type: 'type1' });
});
