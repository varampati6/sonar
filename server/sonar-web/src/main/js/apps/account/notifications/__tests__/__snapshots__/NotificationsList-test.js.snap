// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<tbody>
  <tr>
    <td>
      notification.dispatcher.type1
    </td>
    <td
      className="text-center"
    >
      <Checkbox
        checked={true}
        id="checkbox-io-type1-channel1"
        onCheck={[Function]}
        thirdState={false}
      />
    </td>
    <td
      className="text-center"
    >
      <Checkbox
        checked={false}
        id="checkbox-io-type1-channel2"
        onCheck={[Function]}
        thirdState={false}
      />
    </td>
  </tr>
  <tr>
    <td>
      notification.dispatcher.type2
    </td>
    <td
      className="text-center"
    >
      <Checkbox
        checked={true}
        id="checkbox-io-type2-channel1"
        onCheck={[Function]}
        thirdState={false}
      />
    </td>
    <td
      className="text-center"
    >
      <Checkbox
        checked={true}
        id="checkbox-io-type2-channel2"
        onCheck={[Function]}
        thirdState={false}
      />
    </td>
  </tr>
</tbody>
`;
